### ISSUE TITLE
Does gcc-pru also have support for amd64 arch

### ISSUE THREAD

**Issue by VedantParanjape** [rcn-ee/repos#48]
Debian control file says arch as "any", but in the package repository only armhf package is available.

Is amd64 debian package available for we need to create one ourselves, if yes, could you provide instructions to do so?

Regards,
Vedant Paranjape

_Sent from my moto g(6) using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by dinuxbg** [comment]
Volunteers are welcome to [cleanup](https://github.com/dinuxbg/gnupru/issues/20) and maintain binutils-pru, gcc-pru and gnuprumcu packages in Debian mainline :) Until someone steps forward, people would have to build from source for amd64.

My understanding is that RCN's repos is [only](https://rcn-ee.com/deb/) for armhf. Hence this issue should be closed as "not applicable".

**Comment by VedantParanjape** [comment]
But, we can host a amd64 on launchpad.net ? Can i use the same package generator scripts as used in armhf for amd64 ? I have created a dockerimage with amd64 pru-gcc, but it is too big (4GB), packaged pru-gcc would help a lot.
https://hub.docker.com/repository/docker/simppru/amd64-build-image

Regards,
Vedant

**Comment by dinuxbg** [comment]
You can reuse the debian rules and configs to package for any architecture. I expect them to work for amd64.

The most useful action from a volunteer would be ongoing maintenance in mainline Debian. One-off builds tend to get obsolete quickly and require much more maintenance in the long run.

**Comment by RobertCNelson** [comment]
@VedantParanjape which arm64 board are you working with?

**Comment by VedantParanjape** [comment]
@dinuxbg I am kinda new in this stuff, right now, it will be tough to maintain pru-gcc, if you are willing to point to the right stuff, I'll be happy to contribute. 

I built packages with the given debian rules on amd64 arch. But on installing gnuprumcu pru-gcc and binutils, on running pru-gcc -mmcu=am335x.pru0, i get device spec file not found.

Upon a close inspection of gnuprumcu package, i found it didn't package those files at all.

Did i do something wrong? Or do i also need ti-pru-am335x package?

**Comment by VedantParanjape** [comment]
> @VedantParanjape which arm64 board are you working with?
> 
> nm... amd64, you want a cross build....

Yes, because simppru package that i created also supports amd64, and pru-gcc is a dependency of that.

I built the packages. First ran the download sources script then went into appropriate directory and ran this command: `dpkg-buildpackage -us -uc`

I did this for gnuprumcu, gcc-pru and binutils-pru. Installed it on my system. But pru-gcc doesn't work, it says device spec files missing.

On a closer inspection of gnuprumcu package, it had nothing but docs, no device spec files.

Where did i go wrong?

**Comment by dinuxbg** [comment]
> Where did i go wrong?
I'm not sure why that gives you an empty package. You can try this to build gnuprumcu debian package:

```
git clone https://github.com/dinuxbg/gnuprumcu
cd gnuprumcu
debuild -i -us -uc -b
ls -l ../gnuprumcu_*.deb
```

**Comment by VedantParanjape** [comment]
Great, this did the job
